Ladybird/Qt/BrowserWindow.cpp

```diff
@@ -606,7 +606,14 @@ BrowserWindow::BrowserWindow(Vector<URL::URL> const& initial_urls, IsPopupWindow
         tab.focus_location_editor();
     });
     QObject::connect(m_new_window_action, &QAction::triggered, this, [] {
-        (void)static_cast<Ladybird::Application*>(QApplication::instance())->new_window({});
+        auto& window = static_cast<Ladybird::Application*>(QApplication::instance())->new_window({});
+        auto& tab = window.new_tab_from_url(ak_url_from_qstring(Settings::the()->new_tab_page()), Web::HTML::ActivateTab::Yes);
```

Comment:
this is fun. did someone change the behavior of BrowserWindow so that it doesn't automagically create a new tab? I swear this used to do the expected thing :thinking:

Reply:
FYI, this seems to have regressed as of 5f8d852

Reply:
aha, it's from removing this check:

```
        if (initial_urls.is_empty()) {
            new_tab_from_url(ak_url_from_qstring(Settings::the()->new_tab_page()), Web::HTML::ActivateTab::Yes);
        } else {
            for (size_t i = 0; i < initial_urls.size(); ++i) {
                new_tab_from_url(initial_urls[i], (i == 0) ? Web::HTML::ActivateTab::Yes : Web::HTML::ActivateTab::No);
            }
        }
```

When creating a BrowserWindow from the command line, we insert the new tab page into the vector if it's empty. This happens in.... LibWebView/URL.cpp in `sanitize_urls` . By passing an empty vector to the constructor when calling `LadybirdApplication::new_window(urls, ...)`, we forget to create the new tab page.

@trflynn89

Reply:
I suspect that means that we can hack this up by calling `new_window({ Application::the().chrome_options().new_tab_page() })` rather than asking the QSettings about  it?

Reply:
We used to pass the URL directly from the `new_window` callback before efce3d967183f7071d47f27ce93310662217ef5e. Does whatever that commit fixed break if we start passing it again?

> I suspect that means that we can hack this up by calling new_window({ Application::the().chrome_options().new_tab_page() }) rather than asking the QSettings about it?

If the NTP URL changes, I don't think we yet update the chrome options struct, so we probably do need to go through the settings for now